tests/integration/renderer/test_component_as_prop.py

```diff
@@ -121,6 +121,23 @@ def test_rdcap001_component_as_prop(dash_duo):
                     },
                 ],
             ),
+            ComponentAsProp(
+                dynamic={
+                    "inside-dynamic": Div("dynamic", "inside-dynamic"),
+                    "output-dynamic": Div(id="output-dynamic"),
+                    "clicker": Button("click-dynamic", id="click-dynamic"),
+                    "clicker-dict": Button("click-dict", id="click-dict"),
+                    "clicker-list": Button("click-list", id="click-list"),
+                    "clicker-nested": Button("click-nested", id="click-nested"),
+                },
+                dynamic_dict={"node": {"dict-dyn": Div("dict-dyn", id="inside-dict")}},
+                dynamic_list=[
+                    {"list": Div("dynamic-list", id="inside-list")},
+                ],
+                dynamic_nested_list=[
+                    {"obj": {"nested": Div("nested", id="nested-dyn")}}
+                ],
```

Comment:
Pretty low risk, but if you wanted to cover all bases these props should get two keys & two list items to ensure we're properly looping over everything.